WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -438,10 +448,12 @@ private extension PointOfSaleAggregateModel {
 extension PointOfSaleAggregateModel {
     @MainActor
     func checkOut() async {
+        collectOrderPaymentAnalyticsTracker.trackCheckoutTapped()
         orderStage = .finalizing
         await orderController.syncOrder(for: cart, retryHandler: { [weak self] in
             await self?.checkOut()
         })
+        collectOrderPaymentAnalyticsTracker.trackOrderCreationSuccess()
```

Comment:
I had my doubts if this was the best place to add the order creation success event, I left my comments in the commit 3f9b72b9625c7b36533dd3f3211f5e4d70a83010

Reply:
Thanks for the commit comment. I generally agree with the reasoning. I do have a question though:
> ... and we always create a new order when this happens

Do we? On iOS we don't create a new order unless order items change. I know it's different on Android. Will these events represent the same thing due to these platform differences?

 In `PointOfSaleOrderController` we have a specific `analytics.track(.orderCreationSuccess)` event to track order creation success.

Reply:
If we need, one way to know if the order was actually created is to make `orderController.syncOrder` to return some `Result<SyncOrderState, Error>`. It could return an enum that tells whether the new order was created, didn't change, or error happened.

Reply:
Oh! Interesting, in that case using the `Result` approach feels more adequate here as we need to use the output. I've made the changes on https://github.com/woocommerce/woocommerce-ios/pull/15148/commits/0dd8512368858e46449e5ce9b4afb693845c1aab and added additional tests.